include/constants/battle.h

```diff
@@ -241,18 +240,19 @@
 #define SIDE_STATUS_SCREEN_ANY     (SIDE_STATUS_REFLECT | SIDE_STATUS_LIGHTSCREEN | SIDE_STATUS_AURORA_VEIL)
 
 // Field affecting statuses.
-#define STATUS_FIELD_MAGIC_ROOM                     (1 << 0)
-#define STATUS_FIELD_TRICK_ROOM                     (1 << 1)
-#define STATUS_FIELD_WONDER_ROOM                    (1 << 2)
-#define STATUS_FIELD_MUDSPORT                       (1 << 3)
-#define STATUS_FIELD_WATERSPORT                     (1 << 4)
-#define STATUS_FIELD_GRAVITY                        (1 << 5)
-#define STATUS_FIELD_GRASSY_TERRAIN                 (1 << 6)
-#define STATUS_FIELD_MISTY_TERRAIN                  (1 << 7)
-#define STATUS_FIELD_ELECTRIC_TERRAIN               (1 << 8)
-#define STATUS_FIELD_PSYCHIC_TERRAIN                (1 << 9)
-#define STATUS_FIELD_ION_DELUGE                     (1 << 10)
-#define STATUS_FIELD_FAIRY_LOCK                     (1 << 11)
+#define REMOVE_ALL_TERRAIN                          0 // Just for readability
```

Comment:
I'm not convinced here is the right place for this. It's not a value you'd assign to `gFieldStatuses`.

I think it's a move effect argument? I'm not thrilled about a design where the value is a set of field statuses to apply, except if it's empty and then it's interpreted as clearing a subset of the possible set of field statuses (just the terrains).

Also I'm a little uneasy about `argument` being a `u16`, but `gFieldStatuses` being a `u32`. Hopefully we'd get a compiler warning if we ever extend past the limit of a `u16`; although if you're planning on refactoring arguments to be `u32`s anyway this isn't a concern.

Reply:
> I think it's a move effect argument? I'm not thrilled about a design where the value is a set of field statuses to apply, except if it's empty and then it's interpreted as clearing a subset of the possible set of field statuses (just the terrains).

You're right actually - I think I bent myself backwards of trying to maintain the same effect (as before it was `HIT_SET_REMOVE_TERRAIN`) when I should just have a separate move_effect for that

> if you're planning on refactoring arguments to be u32s anyway this isn't a concern.

🤣